scripts/makekeys

```diff
@@ -10,7 +10,16 @@ pattern = re.compile(r"^#define\s+XKB_KEY_(?P<name>\w+)\s+(?P<value>0x[0-9a-fA-F
 matches = [pattern.match(line) for line in open(sys.argv[1])]
 entries = [(m.group("name"), int(m.group("value"), 16)) for m in matches if m]
 
-entries_isorted = sorted(entries, key=lambda e: e[0].lower())
+# Sort based on the keysym name:
+#   1. Sort by the casefolded name: e.g. kana_ya < kana_YO.
+#   2. If same casefolded name, then sort by cased name, i.e for
+#      ASCII: upper before lower: e.g kana_YA < kana_ya.
+# E.g. kana_YA < kana_ya < kana_YO < kana_yo
+# WARNING: this sort must not be changed, as some functions e.g.
+# xkb_keysym_from_name rely on upper case variant occuring first.
+entries_isorted = sorted(entries, key=lambda e: (e[0].casefold(), e[0]))
```

Comment:
Something I noticed (not intended as a comment on this PR specifically). We currently run this script (through `update-keysyms`) using `export LC_CTYPE=C`.

- I wonder if changing to `LC_CTYPE=C.UTF-8` has the same results, if so I think we should do it since `C` is legacy.

- Since we're sorting, I think we also need to set `LC_COLLATE`, otherwise we might end up using a different locale for the ordering? (E.g. if the developer is Turkish, might have the infamous Turkish-i issue). I'm assuming Python respects `LC_COLLATE` but I'm not sure.

Reply:
Since keysym names are ASCII-only I think is does not matter much. I tried using `tr_TR.UTF-8` and I have seen no difference.

I do not mind to set these though, see the corresponding commit.

Reply:
Thanks!